layouts/partials/sidebar.html

```diff
@@ -3,18 +3,33 @@
 <nav class="section-sidebar">
   <ul class="nav flex-column">
     {{ range $menu }}
-      <li class="nav-item">
-        {{ $isCurrent := and .Page (eq .Page.RelPermalink $.RelPermalink) }}
-        <a class="nav-link {{ if $isCurrent }}active{{ end }}" href="{{ if .Page }}{{ .Page.RelPermalink }}{{ else }}{{ .URL }}{{ end }}">
-          {{ .Name }}
-        </a>
-
-        {{ if and $isCurrent .Page.Params.show_section_nav }}
-          <div class="sidebar-toc nav flex-column ms-4 fs-6">
-            {{ .Page.TableOfContents }}
-          </div>
+    <li class="nav-item">
+      {{ $isCurrent := and .Page (eq .Page.RelPermalink $.RelPermalink) }}
+      <a
+        class="nav-link {{ if eq .Page.RelPermalink $isCurrent }}active{{ end }}"
+        href="{{ .Page.RelPermalink }}"
+      >
+        {{ .Name }}
+      </a>
+      {{ if and $isCurrent .Page.Params.show_section_nav }}
+      <div class="sidebar-toc nav flex-column ms-4 fs-6">
+        {{ .Page.TableOfContents }}
+      </div>
+      {{ else if .HasChildren }}
+      <ul class="nav flex-column ms-4">
+        {{ range .Children }}
+        <li class="nav-item">
+          <a
+            class="nav-link {{ if eq .Page.RelPermalink $isCurrent }}active{{ end }}"
+            href="{{ .Page.RelPermalink }}"
+          >
```

Comment:
For the code, I don't this kinda styling where the closing tag hangs separately 😅 

```
 <a class="nav-link py-1 {{ if eq .Page.RelPermalink $isCurrent }}active{{ end }}"
    href="{{ .Page.RelPermalink }}">
```

Again this is just my personal preference.